<launch>
  <arg name="launch_prefix" default="" /> 
  <arg name="camera_name" default="/camera_rect" />
  <arg name="camera_frame" default="camera" />
  <arg name="image_topic" default="image_rect" />

  <arg name="camera_device" default="/dev/video2" />
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />

  <include file="$(find colab_reachy_camera)/launch/camera_image_and_info.launch">
    <arg name="cam_dev" value="$(arg camera_device)"/>
  </include>

  <!-- Set parameters -->
  <rosparam command="load" file="$(find colab_reachy_grasp)/config/settings.yaml" ns="$(arg node_namespace)"/>
  <rosparam command="load" file="$(find colab_reachy_grasp)/config/tags.yaml" ns="$(arg node_namespace)"/>

  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />
    <param name="publish_tf" type="bool" value='false' />
  </node>

  <node pkg="colab_reachy_grasp" type="apriltag_topic_pub.py" name="apriltag_topic_publisher" output="screen" />

</launch>
